src/main/java/com/aws/greengrass/smbridge/clients/SMClient.java

```diff
@@ -0,0 +1,127 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.smbridge.clients;
+
+import com.amazonaws.greengrass.streammanager.client.StreamManagerClientFactory;
+import com.amazonaws.greengrass.streammanager.client.exception.StreamManagerException;
+import com.amazonaws.greengrass.streammanager.model.MessageStreamDefinition;
+import com.amazonaws.greengrass.streammanager.model.MessageStreamInfo;
+import com.amazonaws.greengrass.streammanager.model.StrategyOnFull;
+import com.amazonaws.greengrass.streammanager.model.export.ExportDefinition;
+import com.aws.greengrass.componentmanager.KernelConfigResolver;
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.smbridge.StreamMessage;
+import com.aws.greengrass.smbridge.StreamDefinition;
+import com.amazonaws.greengrass.streammanager.client.StreamManagerClient;
+
+import javax.inject.Inject;
+import java.util.List;
+
+public class SMClient {
+    private static final Logger LOGGER = LogManager.getLogger(SMClient.class);
+
+    private StreamManagerClient smClient;
+    private MessageStreamDefinition defaultStream;
+    /**
+     * Ctr for SMClient.
+     *
+     * @param topics             topics passed in by Nucleus
+     * @throws SMClientException if unable to create SM Client
+     */
+    @Inject
+    public SMClient(Topics topics) throws SMClientException {
+        this(topics, null);
+        // TODO: Handle the case when serverUri is modified
+        try {
+            this.smClient = StreamManagerClientFactory.standard().build();
+        } catch (StreamManagerException e) {
+            throw new SMClientException("Unable to create a SM client", e);
+        }
+    }
+    protected SMClient(Topics topics, StreamManagerClient streamManagerClient){
+        this.smClient = streamManagerClient;
+        // TODO: Configurable default stream
+        this.defaultStream = new MessageStreamDefinition();
+        this.defaultStream.setName("mqttToStreamDefaultStreamName");
+        this.defaultStream.setTimeToLiveMillis(9223372036854L);
+        this.defaultStream.setStrategyOnFull(StrategyOnFull.RejectNewData);
+    }
+
+    // TODO: Connect to stream manager on custom port
+    //  https://docs.aws.amazon.com/greengrass/v2/developerguide/use-stream-manager-in-custom-components.html
+    public void start() throws SMClientException {
+        try {
+            updateOrCreateStream(defaultStream);
+        } catch (StreamManagerException e) {
+            LOGGER.atError().log("Encountered StreamManagerException while starting SM Client");
+            throw new SMClientException("Unable to start SMClient");
+        }
+    }
+
+    public void publishOnDefaultStream(byte[] payload) throws SMClientException{
+        // TODO: Configurable default stream name
+        String defaultStreamName = "mqttToStreamDefaultStreamName";
```

Comment:
Move to a common place. I see it in multiple places.

Reply:
They are similarly configuring a "default" stream, but it is two different defaults. One is the default stream for reserved topics, and another is the default settings for a stream which the user has configured the bridge to forward towards, but has not yet instantiated with the SM via a separate custom component. Hence the separate logic, since eventually they will each be customized by user config.